### PR TITLE
Game ends when player has no more lives + game over screen and restarting

### PR DESCRIPTION
**Pull request by ivan-BRHM**
#13, #15 
This pr includes quite a lot more features than intended, but despite the features being so intertwined i think it went allright.
These features include:
* Converted game states to views
* Player dies when they have no more lives
* Game over screen/game state
* Restarting the game

### CODE REVIEW

File: my_game.py [hunk 1]
```diff
@@ -297,31 +283,64 @@ def destroy(self):
...
-class MyGame(arcade.Window):
+        self.play_button = None
+        self.title_graphics = None
+
+        arcade.set_background_color(arcade.color.AMAZON)
+
+    def on_show_view(self):
+
+        self.title_graphics = arcade.load_texture("images/UI/asteroidsTitle.png")
+        self.play_button = arcade.load_texture("images/UI/asteroidsStartButton.png")
```
Comment (tobiasbp) [inline my_game.py:303]: Maybe we could set those variables in `__init__`, and then not have `on_show_view()`?

File: my_game.py [hunk 2]
```diff
@@ -419,132 +430,100 @@ def setup(self):
...
+            self.player_sprite.angle+= -PLAYER_ROTATE_SPEED
+
+        # rotate player with joystick if present
+        if self.joystick:
+            self.player_sprite.angle += round(self.joystick.x) * -PLAYER_ROTATE_SPEED
+
+        # checks if ufo shot collides with player
+        if any(self.player_sprite.collides_with_list(self.ufo_shot_list)):
+            self.player_sprite.lives -= 1
+
```
Comment (tobiasbp) [inline my_game.py:489]: Will the player not be hit by the same ufo shot in the next frame?

File: my_game.py [hunk 3]
```diff
@@ -419,132 +430,100 @@ def setup(self):
...
+        for s in self.player_shot_list:
+            for a in arcade.check_for_collision_with_list(s, self.asteroid_list):
+                s.kill()
+                a.kill()
+
+        # check if the player is dead
+        if self.player_sprite.lives <= 0:
+            game_over_view = GameOverView()
+            self.window.show_view(game_over_view)
+
```
Comment (tobiasbp) [inline my_game.py:508]: It would be nice to have the death check as the last thing in `on_update()`

**Review by tobiasbp — CHANGES_REQUESTED** [review]
Looks good. I have some minor requests.

File: my_game.py (hunk 2 above)
Comment (ivan-BRHM) [reply my_game.py:489]: it will. Gonna fix that right after this.

File: my_game.py [hunk 4]
```diff
@@ -117,17 +102,6 @@ def thrust(self):
...
-        player_speed_vector_length = math.sqrt(self.change_x**2 + self.change_y**2)
-
-        # Calculating the value used to lower the players speed while keeping the x - y ratio
-        player_x_and_y_speed_ratio = PLAYER_SPEED_LIMIT/player_speed_vector_length
-
-        # If player is too fast slow it down
-        if player_speed_vector_length > PLAYER_SPEED_LIMIT:
-            self.change_x *= player_x_and_y_speed_ratio
-            self.change_y *= player_x_and_y_speed_ratio
-
```
Comment (tobiasbp) [inline my_game.py:130]: Why are you removing this?

**Review by tobiasbp — CHANGES_REQUESTED** [review]

**Comment by tobiasbp** [comment]
Den klager over brug af ikke-defineret variabel _PLAYER_SPEED_:
https://github.com/tobiasbp/python-arcade-tuesday-02/actions/runs/3311783180/jobs/5467647326

**Comment by ivan-BRHM** [comment]
Ja, den variabel er en gammel en. Den skal bare fjernes helt fra filen - overså den i min conflicts-resolving.

File: my_game.py (hunk 2 above)
Comment (tobiasbp) [reply my_game.py:530]: Was fixed in other branch.

**Review by tobiasbp — APPROVED** [review]
Looks..... good... I guess.... :)

**Pull request merged**